File: CPP/PROGRAM/Monitor.cpp
#pragma once
#include <iostream>
#include <string>
#include "Peripheral.cpp"

using namespace std;

class Monitor : public Peripheral{
    //atribut
    private:
        string resolusi;
        int refreshRateHz;

    public:
        //konstruktor
        Monitor(){
            this->resolusi = "";
            this->refreshRateHz = 0;
        }

        Monitor(string resolusi, int refreshRateHz, string tipePeripheral, string interface, string merk, string nama) : Peripheral(tipePeripheral, interface, merk, nama){
            this->resolusi = resolusi;
            this->refreshRateHz = refreshRateHz;
        }

        //getter dan setter
        void setResolusi(string resolusi){
            this->resolusi = resolusi;
        }

        void setRefreshRateHz(int refreshRateHz){
            this->refreshRateHz = refreshRateHz;
        }

        string getResolusi(){
            return this->resolusi;
        }

        int getRefreshRateHz(){
            return this->refreshRateHz;
        }

        ~Monitor(){

        }
};

File: PYTHON/PROGRAM/Monitor.py
from Peripheral import Peripheral

class Monitor(Peripheral):
    #konstruktor
    def __init__(self, tipePeripheral = "", interface = "", merk = "", nama = "", resolusi = "", refreshRateHz = 0):
        super().__init__(merk, nama, tipePeripheral, interface)
        self.__resolusi = resolusi
        self.__refreshRateHz = refreshRateHz

    #getter dan setter
    def getResolusi(self):
        return self.__resolusi
    
    def setResolusi(self, resolusi):
        self.__resolusi = resolusi

    def getRefreshRateHz(self):
        return self.__refreshRateHz
    
    def setRefreshRateHz(self, refreshRateHz):
        self.__refreshRateHz = refreshRateHz